src/lib.rs

```diff
@@ -535,12 +534,18 @@ note: skipped the following crates since they have no library target: {skipped}"
 
                             if lint_workspace_key || metadata_workspace_key {
                                 if let Some(workspace) = &workspace_overrides {
-                                    overrides.push(Arc::clone(workspace));
+                                    for level in workspace {
+                                        overrides.try_push(level).context("configuration conflict found in workspace overrides.\n\
+                                            use the `priority` field in the manifest to set precedence.")?;
+                                    }
                                 }
                             }
 
                             if let Some(package) = package_overrides {
-                                overrides.push(Arc::new(package.inner));
+                                for level in package.into_stack() {
+                                overrides.try_push(&level).with_context(|| format!("configuration conflict found in package {} overrides.\n\
+                                    use the `priority` field in the manifest to set precedence.", selected.name))?;
+                                }
```

Comment:
Could you add tests that trigger each of these errors? I'm curious to see how they are displayed to users, since I suspect they'll be somewhat common as soon as we add more groups — nobody will remember to add the priority key *until* they see one of these errors.

Reply:
As a side note, maybe these should say "`priority` field in the manifest's lint configuration"? Or is that too long or otherwise undesirable?